backend/src/main/kotlin/com/tngtech/apicenter/backend/domain/entity/Role.kt

```diff
@@ -0,0 +1,22 @@
+package com.tngtech.apicenter.backend.domain.entity
+
+import java.util.*
+
+enum class Role {
+    VIEWER {
+        override fun permissions(): EnumSet<PermissionType> =
+                EnumSet.of(PermissionType.VIEW)
+    },
+
+    VIEWER_X {
+        override fun permissions(): EnumSet<PermissionType> =
+                EnumSet.of(PermissionType.VIEW, PermissionType.VIEWPRERELEASE)
+    },
+
+    EDITOR {
+        override fun permissions(): EnumSet<PermissionType> =
+                EnumSet.of(PermissionType.VIEW, PermissionType.VIEWPRERELEASE, PermissionType.EDIT)
+    };
+
+    abstract fun permissions(): EnumSet<PermissionType>
```

Comment:
Instead of a function, you can define fields for enums:
```
enum class Role(permissions: EnumSety<PermissionType> {
    VIEWER(EnumSet.of(VIEW))
    VIEWER_X(EnumSet.of(VIEW, VIEW_PRERELEASE)
...
```